.buildkite/scripts/pipelines/chromium_linux_build/build.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+CHROMIUM_COMMIT_HASH=$(buildkite-agent meta-data get "chromium_commit_hash")
+
+echo "---Preparing to build Chromium of commit hash: $CHROMIUM_COMMIT_HASH"
+
+BUILD_ROOT_DIR="$HOME/chromium"
+
+KIBANA_CHECKOUT_DIR="$(pwd)"
+
+BUILD_SCRIPT="$KIBANA_CHECKOUT_DIR/x-pack/build_chromium"
+
+# Create a dedicated working directory outside of the default buildkite working directory.
+mkdir "$BUILD_ROOT_DIR" && cd "$BUILD_ROOT_DIR"
+
+ARTIFACT_STAGING_STORAGE_BUCKET="gs://headless_shell_staging"
+ARTIFACT_PROD_STORAGE_BUCKET="gs://headless_shell"
+
+ARTIFACT_QUERY="chromium-${CHROMIUM_COMMIT_HASH:0:7}-.*_$PLATFORM_VARIANT"
+
+## impersonate service account that has access to our storage bucket 
+"$KIBANA_CHECKOUT_DIR/.buildkite/scripts/common/activate_service_account.sh" "kibana-ci-access-chromium-blds"
+
+# Query to determine if expected build artifact from the storage bucket exists
+artifacts=$(gsutil ls "$ARTIFACT_STAGING_STORAGE_BUCKET" | grep "$ARTIFACT_QUERY" || true)
+
+if [[ -z "$artifacts" ]]; then
+  echo "No files found matching the query: $ARTIFACT_QUERY"
+
+   echo "---Chromium build does not exist in the bucket, proceeding with the build"
+
+    # Install the required packages for building chromium
+    sudo apt-get update && \
+    sudo apt-get install -y curl git build-essential pkg-config gcc gperf python3 && \
+    sudo rm -rf /var/lib/apt/lists/*
+
+    BUILD_SCRIPT_SYMLINK="$BUILD_ROOT_DIR/build_scripts"
+
+    # link existing build_chromium directory from kibana
+    ln -s "$BUILD_SCRIPT" "$BUILD_SCRIPT_SYMLINK"
+
+    # Allow our scripts to use depot_tools commands
+    export PATH=$BUILD_ROOT_DIR/depot_tools:$PATH
+
+    # Install the OS packages, configure the environment, download the chromium source (56GB)
+    python3 "$BUILD_SCRIPT_SYMLINK/init.py"
+
+    echo "---Building $PLATFORM_VARIANT Chromium of commit hash: $CHROMIUM_COMMIT_HASH"
+
+    # Run the build script with the path to the chromium src directory, the git commit hash
+    python3 "$BUILD_SCRIPT_SYMLINK/build.py" "$CHROMIUM_COMMIT_HASH" "$PLATFORM_VARIANT"
+
+    echo "---Upload build artefact to prod storage bucket"
```

Comment:
Should we have this upload the build artifact to staging? I guess it doesn't matter very much, since we don't have a strategy of testing the build in staging and then promote it to prod.

Reply:
it actually gets uploaded to staging however it happens implicitly through the `build.py` so it's not so apparent... I'll add a comment so it's much clearer this happens, you also do have a point about not exactly have a strategy for testing the build in staging before uploading it to prod. We upload directly to prod now because whilst building the PR the binaries are fetched from prod, we could however improve this beyond this PR in the future so that for the generated PR builds kibana uses the staging artifacts, and then on merge we upload the artifacts to prod.

Reply:
Having a strategy where we're able to test builds in staging makes sense so that we aren't "polluting" the prod bucket with builds that we aren't sure about. But this need has been around for a long time and discussion about it shouldn't block this PR. Maybe we should just file a followup issue about having a staging test strategy.

Reply:
I think you make a fair point about the pollution of the prod storage bucket, I created an issue here https://github.com/elastic/kibana/issues/220042 to track this.